src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/RuntimeBinder.cs

```diff
@@ -1575,7 +1570,7 @@ internal EXPR BindExplicitConversion(ArgumentObject[] arguments, Type returnType
             int indexOfLast = arguments.Length - 1;
             EXPR rhs = CreateArgumentEXPR(arguments[indexOfLast], locals[indexOfLast]);
 
-            if (arguments[0] == null)
+            if (arguments[0].Type == null)
```

Comment:
This is pretty much equivalent in that `Type` is never set to null, so it can only be null if `arguments[0]` is set to the default value, just as with it being null itself were it still a reference type.

(I think this might never pass and so it's guarding dead code, but I'd want to be 100% sure before cutting it out entirely).